checks/checkforcontent.go

```diff
@@ -99,10 +97,50 @@ func CheckFilesContent(checkName, shellPathFnPattern string,
 			res = false
 		}
 	}
-
 	if res {
 		return checker.MakePassResult(checkName)
 	}
 
 	return checker.MakeFailResult(checkName, nil)
 }
+
+// UPGRADEv2: to rename to CheckFilesContent.
+func CheckFilesContent2(shellPathFnPattern string,
+	caseSensitive bool,
+	c *checker.CheckRequest,
+	onFileContent func(path string, content []byte,
+		dl checker.DetailLogger) (bool, error),
+) (bool, error) {
+	predicate := func(filepath string) bool {
+		// Filter out Scorecard's own test files.
+		if isScorecardTestFile(c.Owner, c.Repo, filepath) {
+			return false
+		}
+		// Filter out files based on path/names using the pattern.
+		b, err := isMatchingPath(shellPathFnPattern, filepath, caseSensitive)
+		if err != nil {
+			return false
```

Comment:
rather than discarding this error, do we need to pipe it through to the caller somehow?